pkg/http/api.go

```diff
@@ -231,6 +231,34 @@ func SetSecrets(host string, verifyTLS bool, apiKey string, project string, conf
 	return computed, Error{}
 }
 
+// SetSecretNote for specified project and config
+func SetSecretNote(host string, verifyTLS bool, apiKey string, project string, config string, secret string, note string) (models.SecretNote, Error) {
+	reqBody := map[string]interface{}{}
+	reqBody["secret"] = secret
+	reqBody["note"] = note
+	body, err := json.Marshal(reqBody)
+	if err != nil {
+		return models.SecretNote{}, Error{Err: err, Message: "Invalid secret note"}
+	}
+
+	var params []queryParam
+	params = append(params, queryParam{Key: "project", Value: project})
+	params = append(params, queryParam{Key: "config", Value: config})
+
+	statusCode, _, response, err := PostRequest(host, verifyTLS, apiKeyHeader(apiKey), "/v3/configs/config/secrets/note", params, body)
```

Comment:
Go has a similar mechanism to JavaScript for URLs that we could leverage:
```go
url, err := url.Parse("https://www.aurl.com")
if err != nil {
	//...
}
values := url.Query()
values.Add("key1", "value1")
values.Add("key2", "value2")
// values.Del
// values.Set
url.RawQuery = values.Encode()

log.Println(url.String())
```

Would this be a worthwhile refactor?

Reply:
Discussed elsewhere: https://github.com/DopplerHQ/cli/pull/351#discussion_r1052816339